filebeat/input/v2/observe.go

```diff
@@ -0,0 +1,45 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package v2
+
+// StatusObserver is used to report a standardized set of state change events
+// of an input.
+type StatusObserver interface {
+	// Starting indicates that the input is about to be configured and started.
+	Starting()
```

Comment:
It can be public or not (when other people will redefine it in their own package)

The observer is also for a developer, but mostly for internal use. Let's say that in the context of the agent we could have a special observer that keeps track of errors in the system and report them to the fleet. 

Other possible builtin/chainable observer
- We could also have a standard reporter that targets the logs.
- One that keeps the local state of the system for the web API that the beat exposes.

So in the sense the StatusObserver is similar to an io.Reader in the usefulness for internal development.

Reply:
@exekias Do you mean the Starting method in particular, or the interface overall?

Reply:
It depends on how you implement the actual input. We provide some helpers, but the most bare bones implementation would be compatible to Runner as is used by autodiscovery (this is, so we can integrate with reloading + autodiscovery).